libqtile/utils.py

```diff
@@ -420,13 +424,37 @@ async def add_signal_receiver(
         "/org/freedesktop/DBus",
         "AddMatch",
         "s",
-        rule,
+        [rule],
     )
 
     # Check if message sent successfully
     if msg and msg.message_type == MessageType.METHOD_RETURN:
         bus.add_message_handler(callback)
+        dbus_match_calls.append((bus, rule))
         return True
 
     else:
         return False
+
+
+async def remove_signal_receiver(bus: MessageBus, rule: str) -> bool:
+    logger.debug("Removing dbus match rule: %s", rule)
+    msg = await bus.call(
+        Message(
+            message_type=MessageType.METHOD_CALL,
+            destination="org.freedesktop.DBus",
+            interface="org.freedesktop.DBus",
+            path="/org/freedesktop/DBus",
+            member="RemoveMatch",
+            signature="s",
+            body=[rule],
+        )
+    )
+
+    bus.disconnect()
```

Comment:
This may be a bad idea as there could be multiple match rules per bus. So we either want a dictionary with the bus as the key and a list of rules. Or... is disconnecting the bus sufficient to delete the remove the rules?